framework/assets/yii.gridView.js

```diff
@@ -118,6 +118,10 @@
 
             var pos = settings.filterUrl.indexOf('?');
             var url = pos < 0 ? settings.filterUrl : settings.filterUrl.substring(0, pos);
+            var hashPos = settings.filterUrl.indexOf('#');
+            if (hashPos >= 0) {
+                url += settings.filterUrl.substring(pos);
```

Comment:
`hashPos` must be used instead of a `pos` here. Also the anchor needs to be added only when get parameters was cut off, otherwise url already contains anchor and it will lead to duplication again.